src/WooCommerce.php

```diff
@@ -57,30 +57,55 @@ public function reviewsTemplate(string $template): string
             return $template;
         }
 
-        return $this->template($template);
+        return $this->templatePart($template);
+    }
+
+    /**
+     * Add blade templates for the woocommerce status page if user is admin and
+     * loaded wc status page.
+     */
+    public function getTemplate(string $template, string $templateName, array $args): string
+    {
+        $themeTemplate = $this->locateThemeTemplate($templateName);
+
+        // return theme filename for status screen
+        if (is_admin() &&
+            !wp_doing_ajax() &&
+            get_current_screen() &&
+            get_current_screen()->id === 'woocommerce_page_wc-status') {
+            return $themeTemplate ?: $template;
+        }
+
+        // return default template, output already rendered by "templatePart" method.
+        return $template;
```

Comment:
Why exactly is all this needed? Cant we just pass the previous `template()` method to the `wc_get_template` filter?

Reply:
This method returns template name in several hooks in WC source, one of them is on ```wc status``` page, that shows all overriden templates of theme.
in previous ```sage-woocommerce``` package this hook was existed to show the files and versions of files, that we have now added.

Reply:
Ah so this would somehow show the templates real name rather than the hashed filename that blade actually uses? i.e. `single-product.blade.php` instead of `d63d0e21fdc05f618d55ef306c54af82.php`? That would indeed be useful

Reply:
> Ah so this would somehow show the templates real name rather than the hashed filename that blade actually uses? i.e. `single-product.blade.php` instead of `d63d0e21fdc05f618d55ef306c54af82.php`? That would indeed be useful

Yes, I've tested this part, also if we have an outdated files,it will be noticed there because the version of the files is read from their compiled files. (sorry for my bad english).